File: simple-record-app/src/Services/AddressPersonService.ts
import { IAddressPersonService } from '../core/Contracts/IAddressPersonService';
import { CreateAddressPersonInputModel } from '../core/InputModels/CreateAddressPersonInputModel';
import { DeleteAddressPersonInputModel } from '../core/InputModels/DeleteAddressPersonInputModel';
import { GenericServiceResult } from '../core/ViewModels/GenericServiceResult';
import HttpServer from './HttpServer';

class AddressPersonService implements IAddressPersonService {
  private http: HttpServer;

  constructor() {
    this.http = new HttpServer();
  }

  public async CreateAddressPerson(data: CreateAddressPersonInputModel): Promise<GenericServiceResult> {
    const response = await this.http.post<GenericServiceResult>('/api/AddressPerson/CreateAddressPerson', data);
    return response;
  }

  public async DeleteAddressPerson(data: DeleteAddressPersonInputModel): Promise<GenericServiceResult> {
    const response = await this.http.delete<GenericServiceResult>('/api/AddressPerson/DeleteAddressPerson', data );
    return response;
  }
}

export default AddressPersonService;

File: simple-record-ws/Simple-Record.Application/Services/AddressPersonService.cs
using simple_record.service.InputModels;
using simple_record.service.Repositories;
using simple_record.service.Services.Contracts;

namespace simple_record.service.Services
{
    public class AddressPersonService : IAddressPersonService
    {   
        private readonly IPersonAddressRepository _repository;
        private readonly IPersonRepository _personRepository;

        public AddressPersonService(IPersonAddressRepository repository, IPersonRepository personRepository)
        {
            _repository = repository;
            _personRepository = personRepository;
        }

        public async Task<GenericServiceResult> CreateAddressPerson(CreateAddressPersonInputModel model)
        {
            var getPersonByIdInputModel = new GetPersonByIdInputModel()
            {
                Id = model.PersonId
            };
            var person = await _personRepository.GetPersonByIdAsync(getPersonByIdInputModel);

            if (person == null)
            {
                return new GenericServiceResult("Invalid Id", false, null, null);
            }
            var data = model.ToEntity();
            if (!data.Valid)
            {
                return new GenericServiceResult("Invalid data", false, null, data.Notifications);
            }

            await _repository.AddPersonAddressAsync(model);

            return new GenericServiceResult("Created AddressPerson", true, data, data.Notifications);
        }

        public async Task<GenericServiceResult> DeleteAddressPerson(DeleteAddressPersonInputModel model)
        {
            var getPersonByIdInputModel = new GetPersonByIdInputModel()
            {
                Id = model.PersonId
            };
            var person = await _personRepository.GetPersonByIdAsync(getPersonByIdInputModel);

            if (person == null)
            {
                return new GenericServiceResult("Invalid Id", false, null, null);
            }

            await _repository.DeletePersonAddressAsync(model);

            return new GenericServiceResult("Deleted AddressPerson", true, null, null);
        }
    }
}
